test/WebJobs.Script.Tests/Managment/WebFunctionsManagerTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Tests.Helpers;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Azure.WebJobs.Script.WebHost.Management;
+using Microsoft.Extensions.Logging;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Managment
+{
+    public class WebFunctionsManagerTests : IDisposable
+    {
+        [Fact]
+        public async Task VerifyDurableTaskHubNameIsAdded()
+        {
+            // Setup
+            var settings = CreateWebSettings();
+            var fileSystem = CreateFileSystem(settings.ScriptPath);
+            var loggerFactory = CreateLoggerFactory();
+            var webManager = new WebFunctionsManager(settings, loggerFactory);
+            var httpClient = CreateHttpClient();
+
+            FileUtility.Instance = fileSystem;
+            HttpClientUtility.Instance = httpClient;
+
+            Environment.SetEnvironmentVariable("WEBSITE_AUTH_ENCRYPTION_KEY", SimpleWebTokenTests.GenerateKeyHexString());
+            Environment.SetEnvironmentVariable("WEBSITE_SITE_NAME", "appName");
+
+            // Act
+            (var success, var error) = await webManager.TrySyncTriggers();
+
+            // Assert
+            Assert.True(success, "SyncTriggers should return success true");
+            Assert.True(string.IsNullOrEmpty(error), "Error should be null or empty");
+        }
+
+        private static HttpClient CreateHttpClient()
+        {
+            return new HttpClient(new MockHttpHandler());
+        }
+
+        private static WebHostSettings CreateWebSettings()
+        {
+            return new WebHostSettings
+            {
+                ScriptPath = @"x:\root",
+                IsAuthDisabled = false,
+                IsSelfHost = false,
+                LogPath = @"x:\tmp\log",
+                SecretsPath = @"x:\secrets",
+                TestDataPath = @"x:\test"
+            };
+        }
+
+        private static IFileSystem CreateFileSystem(string rootPath)
+        {
+            var fileSystem = Substitute.For<IFileSystem>();
+            fileSystem.File.Exists(Arg.Is(Path.Combine(rootPath, "host.json"))).Returns(true);
```

Comment:
you use `Moq` instead of `NSubstitute`. I think `NSubstitute` has a much simpler syntax than `Moq`, but let me know if you'd rather only use one. I can update the mock to use Moq mocks instead.

Reply:
Somehow I managed to miss this PR review. I'd like to hear @mathewc 's take, but I would rather stick to a single mocking framework. I know this is merged, so perhaps just create an issue to track for now.

Reply:
I agree that we should stick to one (Moq) in this repo.

Reply:
the tests aren't actually from any other repo. The other repo that has the same functionality is Kudu and kudu uses Moq as well. I just prefer NSubstitute is all :) 

Sorry I pinged on the PR a couple of times and only Chris looked at it, so I thought I'd just merge it. I'll send another PR to address comments here